fabric-biomes-v1/src/main/java/net/fabricmc/fabric/mixin/biome/MixinAddRiversLayer.java

```diff
@@ -18,42 +18,39 @@
 
 import java.util.Map;
 
-import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
-import net.minecraft.util.registry.BuiltinRegistries;
+import net.minecraft.util.registry.RegistryKey;
 import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.Biomes;
+import net.minecraft.world.biome.BuiltInBiomes;
 import net.minecraft.world.biome.layer.AddRiversLayer;
 import net.minecraft.world.biome.layer.util.LayerRandomnessSource;
 import net.minecraft.world.biome.layer.util.LayerSampler;
 
 import net.fabricmc.fabric.api.biomes.v1.OverworldBiomes;
 import net.fabricmc.fabric.impl.biome.InternalBiomeData;
+import net.fabricmc.fabric.impl.biome.InternalBiomeUtils;
 
 /**
- * Sets river biomes specified with {@link OverworldBiomes#setRiverBiome(Biome, Biome)}.
+ * Sets river biomes specified with {@link OverworldBiomes#setRiverBiome(RegistryKey, RegistryKey)}.
  */
 @Mixin(AddRiversLayer.class)
 public class MixinAddRiversLayer {
-	@Shadow
-	@Final
-	private static int RIVER_ID;
-
 	@Inject(at = @At("HEAD"), method = "sample", cancellable = true)
 	private void sample(LayerRandomnessSource rand, LayerSampler landSampler, LayerSampler riverSampler, int x, int z, CallbackInfoReturnable<Integer> info) {
 		int landBiomeId = landSampler.sample(x, z);
-		Biome landBiome = BuiltinRegistries.BIOME.get(landBiomeId);
+		RegistryKey<Biome> landBiomeKey = Biomes.fromRawId(landBiomeId);
 
 		int riverBiomeId = riverSampler.sample(x, z);
-		Map<Biome, Biome> overworldRivers = InternalBiomeData.getOverworldRivers();
+		Map<RegistryKey<Biome>, RegistryKey<Biome>> overworldRivers = InternalBiomeData.getOverworldRivers();
 
-		if (overworldRivers.containsKey(landBiome) && riverBiomeId == RIVER_ID) {
-			Biome riverBiome = overworldRivers.get(landBiome);
-			info.setReturnValue(riverBiome == null ? landBiomeId : BuiltinRegistries.BIOME.getRawId(riverBiome));
+		if (overworldRivers.containsKey(landBiomeKey) && Biomes.fromRawId(riverBiomeId) == BuiltInBiomes.RIVER) {
```

Comment:
I would suggest... Ok, I think you get the point.

Reply:
Answered above.